src/app/_components/Category.tsx

```diff
@@ -1,60 +1,61 @@
 'use client'
 
-import { useState } from 'react'
+import { useState, useEffect } from 'react'
 
-import type { CategoryOption, CategoryId } from '@/types/categoryTypes'
-
-import { DEFAULT_CATEGORY } from '@/constants/categoryConstants'
+import { ALL_CATEGORIES_OPTION } from '@/constants/filterConstants'
 import { CATEGORY_KEY } from '@/constants/searchParams'
 
 import { useCategory } from '@/hooks/useCategory'
 import { useUpdateSearchParams } from '@/hooks/useUpdateSearchParams'
 
-import { CategoryListbox } from '@/components/CategoryListbox'
 import { CategorySidebar } from '@/components/CategorySidebar'
+import { FilterListbox } from '@/components/FilterListbox'
+import type { OptionType } from '@/components/ListboxOption'
 
 type CategoryProps = {
   query: ReturnType<typeof useCategory>['categoryQuery']
-  options: Array<CategoryOption>
-  counts?: ReturnType<typeof useCategory>['categoryCounts']
+  options: Array<OptionType>
 }
 
-export function Category({ query, options, counts }: CategoryProps) {
-  const [categoryId, setCategoryId] = useState<CategoryId>(
-    query || DEFAULT_CATEGORY,
+export function Category({ query, options }: CategoryProps) {
+  const [categoryId, setCategoryId] = useState<OptionType['id']>(
+    query || ALL_CATEGORIES_OPTION.id,
   )
   const { updateSearchParams, resetSearchParams } = useUpdateSearchParams()
 
-  const selectedCategory = options.find((option) => option.id === categoryId)
+  const selectedCategory =
+    options.find((option) => option.id === categoryId) || ALL_CATEGORIES_OPTION
+
+  useEffect(() => {
+    setCategoryId(query || ALL_CATEGORIES_OPTION.id)
+  }, [query])
```

Comment:
Why is this needed?

`updateCategoryAndParams` already takes care of updating the state no?

Reply:
you're right! think it was a leftover of something I tried

Reply:
✅